### PR TITLE
[createReactClass] remove createReactClass from ToolbarAndroid/ToolbarAndroid.android.js

### PR DESCRIPTION
**Pull request by nd-02110114**
Related to #21581 .
Removed createReactClass from the Libraries/Components/ToolbarAndroid/ToolbarAndroid.android.js

## Test Plan
- [x] npm run prettier
- [x] npm run flow-check-ios
- [x] npm run flow-check-android
## Release Notes
[GENERAL] [ENHANCEMENT] [Libraries/Components/ToolbarAndroid/ToolbarAndroid.android.js] - remove createReactClass dependency

### CODE REVIEW

File: Libraries/Components/ToolbarAndroid/ToolbarAndroid.android.js [hunk 1]
```diff
@@ -63,106 +55,119 @@ const optionalImageSource = PropTypes.oneOfType([
...
+  navIcon?: ?ImageSource,
+  /**
+   * Callback that is called when an action is selected. The only argument that is passed to the
+   * callback is the position of the action in the actions array.
+   */
+  onActionSelected?: ?Function,
+  /**
+   * Callback called when the icon is selected.
+   */
+  onIconClicked?: ?Function,
```
Comment (RSNara) [inline Libraries/Components/ToolbarAndroid/ToolbarAndroid.android.js]: This should be:

```JavaScript
onIconClicked?: ?() => void,
```

File: Libraries/Components/ToolbarAndroid/ToolbarAndroid.android.js [hunk 2]
```diff
@@ -63,106 +55,119 @@ const optionalImageSource = PropTypes.oneOfType([
...
+  logo?: ?ImageSource,
+  /**
+   * Sets the navigation icon.
+   */
+  navIcon?: ?ImageSource,
+  /**
+   * Callback that is called when an action is selected. The only argument that is passed to the
+   * callback is the position of the action in the actions array.
+   */
+  onActionSelected?: ?Function,
```
Comment (RSNara) [inline Libraries/Components/ToolbarAndroid/ToolbarAndroid.android.js]: This should be:

```JavaScript
onActionSelected?: ?(ToolbarAndroidChangeEvent) => void
```

File: Libraries/Components/ToolbarAndroid/ToolbarAndroid.android.js [hunk 3]
```diff
@@ -63,106 +55,119 @@ const optionalImageSource = PropTypes.oneOfType([
...
+    if (position === -1) {
+      this.props.onIconClicked && this.props.onIconClicked();
+    } else {
+      this.props.onActionSelected && this.props.onActionSelected(position);
+    }
+  };
+
+  render() {
     const nativeProps = {
       ...this.props,
```
Comment (RSNara) [inline Libraries/Components/ToolbarAndroid/ToolbarAndroid.android.js]: nit: It looks like `onIconClicked` and `onActionSelected` aren't native props.

```JavaScript
const {
  onIconClicked,
  onActionSelected,
  ...nativeProps,
} = this.props;
```

**Comment by empyrical** [comment]
This used to use NativeMethodsMixin, so you should forward a ref to the native component with this:

https://github.com/facebook/react-native/blob/master/Libraries/Utilities/setAndForwardRef.js

**Comment by empyrical** [comment]
Actually, on closer inspection this class doesn't need setAndForwardRef. Should just need to forward the ref

File: Libraries/Components/ToolbarAndroid/ToolbarAndroid.android.js [hunk 4]
```diff
@@ -195,17 +202,8 @@ const ToolbarAndroid = createReactClass({
...
-    const position = event.nativeEvent.position;
-    if (position === -1) {
-      this.props.onIconClicked && this.props.onIconClicked();
-    } else {
-      this.props.onActionSelected && this.props.onActionSelected(position);
-    }
-  },
-});
+  }
+}
```
Comment (RSNara) [inline Libraries/Components/ToolbarAndroid/ToolbarAndroid.android.js]: To successfully get rid of `NativeMethodMixin`, you need to do this:

```JavaScript
const ToolbarAndroidWithRef = React.forwardRef((props: Props, ref: React.Ref<typeof NativeToolbar>) => {
  return <ToolbarAndroid {...props} forwardedRef={ref} />
});
```

And then inside `ToolbarAndroid`, you need to do this:
```JavaScript
type Props = $ReadOnly<{|
  // ...
  forwardedRef: React.Ref<typeof NativeToolbar>,
|}>

class ToolbarAndroid extends React.Component<Props> {
  // ...
  render() {
    const {
      onIconClicked,
      onActionSelected,
      forwardedRef: ref,
      ...nativeProps,
    } = this.props;    
    // ...

    return <NativeToolbar onSelect={this._onSelect} {...nativeProps} ref={ref}/>
  }
}
```

I think React.forwardRef isn't flow typed yet, so you'll have to find a workaround for that. There are other PRs removing NativeMethodsMixin that accomplish this (e.g: #21885)

Comment (empyrical) [reply Libraries/Components/ToolbarAndroid/ToolbarAndroid.android.js]: The bit that needs to be copied for now specifically is the FlowFixMe comment above the forwardref call, to suppress the type errors

Comment (nd-02110114) [reply Libraries/Components/ToolbarAndroid/ToolbarAndroid.android.js]: 👍

**Comment by nd-02110114** [comment]
CI doesnot work ...

**Comment by nd-02110114** [comment]
@empyrical 
CI passed 🎉

**Comment by nd-02110114** [comment]
@empyrical @RSNara 
Thank you for your patience 🙇

**Pull request closed without merging**